core/src/main/java/com/sequenceiq/cloudbreak/service/stack/CloudResourceAdvisor.java

```diff
@@ -189,7 +191,8 @@ private boolean isThereMasterComponents(ClusterManagerType clusterManagerType, S
     }
 
     private VmType getDefaultVmType(String availabilityZone, CloudVmTypes vmtypes) {
-        VmType vmType = vmtypes.getDefaultCloudVmResponses().get(availabilityZone);
+        VmType vmType = StringUtils.isNotBlank(availabilityZone) ? vmtypes.getDefaultCloudVmResponses().get(availabilityZone)
+                : vmtypes.getDefaultCloudVmResponses().values().iterator().next();
```

Comment:
Instead of null you might get a NoSuchElementException from iterator().next() method call